File: TypeScript/78.子集.ts
function subsets(nums: number[]): number[][] {
    const len = nums.length
    const maxn = Math.floor(Math.pow(2, len))
    const ans: Array<Array<number>> = [...Array(maxn)].map(_ => Array())
    for (let i = 0; i < maxn; ++i)
        for (let j = 0; j < len; ++j)
            if (((i >> j) & 1) == 1)
                ans[i].push(nums[j])
    return ans
};

File: Golang/78.子集.go
package leetcode

import "math"

func subsets(nums []int) [][]int {
	l := len(nums)
	maxn := int(math.Pow(float64(2), float64(l)))
	ans := make([][]int, maxn, maxn)
	for i := 0; i < maxn; i++ {
		for j := 0; j < l; j++ {
			if ((i >> j) & 1) == 1 {
				ans[i] = append(ans[i], nums[j])
			}
		}
	}
	return ans
}
